From inaky@linux.intel.com  Wed Jul 18 15:03:54 2007
From: Inaky Perez-Gonzalez <inaky@linux.intel.com>
Date: Wed, 18 Jul 2007 14:35:50 -0700
Subject: usb: usb_probe_interface() obeys authorization
To: gregkh@kroah.com, "Alan Stern" <stern@rowland.harvard.edu>, "David Brownell" <david-b@pacbell.net>, linux-usb-devel@lists.sourceforge.net
Message-ID: <20070718214243.348596882@sodium.jf.intel.com>


If called and the device is not authorized to be used, it won't
configure the interface and print a message saying so.

Signed-off-by: Inaky Perez-Gonzalez <inaky@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/driver.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -201,6 +201,11 @@ static int usb_probe_interface(struct de
 
 	intf = to_usb_interface(dev);
 	udev = interface_to_usbdev(intf);
+
+ 	if (udev->authorized == 0) {
+ 		dev_err(&intf->dev, "Device is not authorized for usage\n");
+ 		return -ENODEV;
+ 	}
 
 	id = usb_match_id(intf, driver->id_table);
 	if (!id)
